astropy/units/format/vounit.py

```diff
@@ -222,8 +222,12 @@ def _fix_deprecated(cls, x: str) -> list[str]:
         )
 
     @classmethod
-    def _deprecated_unit_warning_message(cls, unit: str) -> str:
-        return (
-            super()._deprecated_unit_warning_message(unit)
-            + f" Suggested: {cls.to_string(cls._units[unit]._represents)}."
-        )
+    def _validate_unit(cls, unit: str, detailed_exception: bool = True) -> UnitBase:
+        if unit in cls._deprecated_units:
+            warnings.warn(
```

Comment:
Is there a reason not to use the `message, category` arg form?

Reply:
Python would be fine with
```python
                f"The unit '{unit}' has been deprecated in the VOUnit standard."
                f" Suggested: {cls.to_string(cls._units[unit]._represents)}.",
                UnitsWarning,
```
but I don't like that because it would be too similar to
```python
                f"The unit '{unit}' has been deprecated in the VOUnit standard.",
                f" Suggested: {cls.to_string(cls._units[unit]._represents)}.",
                UnitsWarning,
```
which has a very different meaning. It would be less confusing to write
```python
                (
                    f"The unit '{unit}' has been deprecated in the VOUnit standard."
                    f" Suggested: {cls.to_string(cls._units[unit]._represents)}."
                ),
                UnitsWarning,
```
but then we might as well save a line (and two commas) by writing the code that I've actually pushed.